src/compiler/core.ts

```diff
@@ -329,6 +329,21 @@ namespace ts {
         return undefined;
     }
 
+    /**
+     * Like `forEach`, but iterates in reverse order.
+     */
+    export function forEachRight<T, U>(array: readonly T[] | undefined, callback: (element: T, index: number) => U | undefined): U | undefined {

```

Comment:
This is probably just a convention I don't recognize, but why "right", rather than "reverse"?

Reply:
Yep, I was modeling off of `Array.prototype.reduceRight`. Without a precedent of convention I would avoid using “handed” language as I imagine it’s somewhat dissonant to folks whose first written language is RTL.